src/torrent.cpp

```diff
@@ -2603,12 +2597,17 @@ bool is_downloading_state(int const st)
 		// argument in the announce, this will make the DHT node use
 		// our source port in the packet as our listen port, which is
 		// likely more accurate when behind a NAT
-		if (settings().get_bool(settings_pack::enable_incoming_utp))
+		// don't set this for SSL torrents because peers need
+		// to connect to the SSL listen port
+		if (settings().get_bool(settings_pack::enable_incoming_utp)
+			&& !is_ssl_torrent())
 			flags |= dht::dht_tracker::flag_implied_port;
 
+		if (is_ssl_torrent())
+			flags |= dht::dht_tracker::flag_ssl_torrent;
```

Comment:
does this do the correct thing for SSL torrents where `settings_pack::enable_incoming_utp` is set to `false`? It probably doesn't matter that much though.

It seems like there may be a simpler way of phrasing this logic. Perhaps by checking `is_ssl_torrent()` first, something along these lines:

```
if (is_ssl_torrent())
{
   flags |= dht::dht_tracker::flag_ssl_torrent;
}
else
{
   if (settings().get_bool(settings_pack::enable_incoming_utp))
      flags |= dht::dht_tracker::flag_implied_port;
}
```